LanguageFeatures/Patterns/logical_and_A06_t01.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// logicalAndPattern ::= ( logicalAndPattern '&&' )? relationalPattern
+///
+/// A pair of patterns separated by & matches only if both subpatterns match.
+/// Unlike logical-or patterns, the variables defined in each branch must not
+/// overlap, since the logical-and pattern only matches if both branches do and
+/// the variables in both branches will be bound.
+///
+/// If the left branch does not match, the right branch is not evaluated. This
+/// only matters because patterns may invoke user-defined methods with visible
+/// side effects.
+///
+/// @description Checks that if the left branch does not match, the right branch
+/// is not evaluated
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "patterns_lib.dart";
+import "../../Utils/expect.dart";
+
+main() {
+  Shape shape1 = Circle(1);
+  switch (shape1) {
+    case Circle(area: 2) && Circle(size: 1):
+      Expect.fail("Pattern should not match");
+      break;
+    default:
+  }
+  Expect.equals("Circle.area=2;", shape1.log);
+
+  Shape shape2 = Square(1);
+  switch (shape2) {
+    case Square(area: 1) && Square(area: 2) && Square(size: 1):
+      Expect.fail("Pattern should not match");
+      break;
+    default:
+  }
+  Expect.equals("Square.area=1;Square.area=2;", shape2.log);
```

Comment:
Great! Thank you for noticing! `shape2.area` getter returns `Unit` object that will be cached. But this object has logger enabled, so at each comparison it will call `_log("=$other;");`. So here and in other places we should expect 
`Expect.equals("Square.area=1;=2;", shape2.log);`

Reply:
I think all these things are working now. Resolving a bunch of threads.